assignment-02/queries.sql

```diff
@@ -0,0 +1,558 @@
+from
+    orders
+        left join
+    orderdetails ON orderdetails.OrderID = orders.OrderID
+        inner join
+    products ON products.ProductID = orderdetails.ProductID
+where
+    products.ProductName in ('Gorgonzola Telino' , 'Gnocchi di nonna Alice',
+        'Raclette Courdavault',
+        'Camembert Pierrot')
+        and orders.OrderDate between '1997-01-01' and '1997-01-31';
+        
+-- 12
+-- What is the full name of the employees who ordered Tofu between 13th of January,1997 and 30th of January,1997
+select distinct
+    concat_ws(' ',
+            employee.FirstName,
+            employee.LastName) as employee_name
+from
+    orders
+        join
+    employee ON orders.EmployeeID = employee.EmployeeID
+        join
+    orderdetails ON orderdetails.OrderID = orders.OrderID
+where
+    orderdetails.ProductID = (select 
+            ProductID
+        from
+            products
+        where
+            ProductName = 'Tofu')
+        and orders.OrderDate between '1997-01-13' and '1997-01-30';
+            
+-- 13
+/* What is the age of the employees in days, months and years who placed orders during the month of
+August. Get employeeID and full name as well */
+select distinct
+    (employee.EmployeeID),
+    concat_ws(' ',
+            employee.FirstName,
+            employee.LastName) as employee_full_name,
+    timestampdiff(day,
+        employee.BirthDate,
+        current_date) as age_in_days,
+    timestampdiff(month,
+        employee.BirthDate,
+        current_date) as age_in_months,
+    timestampdiff(year,
+        employee.BirthDate,
+        current_date) as age_in_years
+from
+    orders
+        inner join
+    employee ON employee.EmployeeID = orders.EmployeeID
+where
+    month(orders.OrderDate) = 08;
+    
+-- 14
+-- Get all the shipper's name and the number of orders they shipped
+select 
+    shippers.CompanyName,
+    count(shippers.ShipperID) as number_of_orders
+from
+    orders
+        join
+    shippers ON shippers.ShipperID = orders.ShipperID
+group by shippers.ShipperID;
+
+-- 15
+-- Get the all shipper's name and the number of products they shipped.
+select 
+    shippers.CompanyName,
+    sum(orderdetails.Quantity) as number_of_products
+from
+    orders
+        join
+    shippers ON orders.ShipperID = shippers.ShipperID
+        join
+    orderdetails ON orderdetails.OrderID = orders.OrderID
+group by shippers.ShipperID;
+
+-- 16
+-- Which shipper has bagged most orders. Get the shipper's id, name and the number of orders
+select 
+    shippers.ShipperID,
+    count(shippers.ShipperID) as number_of_orders
+from
+    orders
+        join
+    shippers ON shippers.ShipperID = orders.ShipperID
+group by shippers.ShipperID
+order by number_of_orders desc
+limit 1;
+
+-- 17
+/* Which shipper supplied the most number of products between 10th August,1996 and 20th
+September,1998. Get the shipper's name and the number of products. */
+select 
```

Comment:
1. limit 0,1
2. you can use date types in between clause

Reply:
1. join started with shippers table
2. using limit 0,1